targets/targets.json

```diff
@@ -4035,6 +4035,48 @@
         "release_versions": ["2", "5"],
         "device_name": "STM32F407VG"
     },
+    "OLIMEX_STM32E407_F407ZG": {
+        "inherits": ["FAMILY_STM32"],
+        "core": "Cortex-M4F",
+        "extra_labels_add": [
+            "STM32F4",
+            "STM32F407",
+            "STM32F407xG",
+            "STM32F407ZG",
+            "STM_EMAC"
+        ],
+        "supported_toolchains": ["ARM", "uARM", "GCC_ARM", "IAR"],
+        "config": {
+            "clock_source": {
+                "help": "Mask value : USE_PLL_HSE_EXTC (need HW patch) | USE_PLL_HSE_XTAL | USE_PLL_HSI",
+                "value": "USE_PLL_HSE_XTAL|USE_PLL_HSI",
+                "macro_name": "CLOCK_SOURCE"
+            },
+            "hse_value": {
+                "macro_name": "HSE_VALUE",
+                "value": "12000000"
+            }
+        },
+        "macros_add": ["USB_STM_HAL"],
+        "overrides": { "lse_available": 0 },
+        "device_has_add": [
+            "ANALOGOUT",
+            "TRNG",
+            "FLASH",
+            "MPU",
+            "EMAC",
+            "CAN",
+            "USBDEVICE"
+        ],
+        "release_versions": ["5"],
+        "device_name": "STM32F407ZG",
+        "target_overrides": {
+          "*": {
```

Comment:
That config doesn't belong to the target definition, please move it to `platform/mbed_lib.json`

Reply:
I removed the target override for stdio baud rate as requested - the first two lines (release_versions, device_name) match other entries so I guess they are fine?